classes/plugins/PluginHelper.php

```diff
@@ -100,10 +100,9 @@ private function extractPlugin(string $filePath, string $originalFileName): stri
      */
     public function installPlugin(string $path, string $originalFileName): Version
     {
-        $fileManager = new FileManager();
-        $sourcePath = $this->extractPlugin($path, $originalFileName);
-        try {
-            $versionFile = $sourcePath . static::PLUGIN_VERSION_FILE;
+        return $this->extractPlugin($path, $originalFileName, function (string $pluginFolder): Version {
```

Comment:
Just enclosed the extraction/cleanup process in a single place and processed through a callback, looks cleaner/safer for me.